plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -734,49 +870,83 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
                 Optional.ofNullable(handle.getCatalogName()),
                 Optional.ofNullable(handle.getSchemaName()),
                 handle.getTableName());
-        String insertSql = buildInsertSql(session, targetTable, temporaryTable, handle.getColumnNames());
+
+        // We conditionally create more than the one table, so keep a list of the tables that need to be dropped.
+        Closer closer = Closer.create();
+        closer.register(() -> dropTable(session, temporaryTable));
 
         try (Connection connection = getConnection(session, handle)) {
+            String columns = handle.getColumnNames().stream()
+                    .map(this::quoted)
+                    .collect(joining(", "));
+
+            String insertSql = format("INSERT INTO %s %s (%s) SELECT %s FROM %s temp_table",
+                    quoted(targetTable),
+                    isTableLockNeeded(session) ? "WITH (TABLOCK)" : "", // TABLOCK is a prerequisite for minimal logging in SQL Server
```

Comment:
I am not a big fan of exposing SQL server quirks in BaseJdbcClient. Can we keep previous approach where we had a protected method to construct whole INSERT SQL?

Reply:
@losipiuk i _definitely_ do not want a protected method responsible for constructing the _whole_ insert sql (i.e. including the semi-join), as that would require the SQL server client to copy a significant amount of implementation, and to pass the `Closer` and `handle.getPageSinkIdColumnName()` to the protected method.

I understand not wanting to have sql server quirks in BaseJdbcClient, but i think there's a better compromise. What if there was a `finishInsertTableIntoClause` (or something like that) method, which would by default return `quoted(targetTable)`, that SqlServerClient would override and add the `TABLOCK` portion?

Reply:
The other option would be the protected method only do the base `INSERT`, and BaseJdbcClient still adds the semi-join changes, but then there's the tricky part about the table alias and keeping the sql server client compatible with the rest of BaseJdbcClient (which is also why I got rid of the protected method)